Patterns restructure (avoid Match/path mode terminology outside Reference section) (#1526)
neo4j/docs-cypher: modules/ROOT/pages/patterns/acyclic-patterns.adoc

:description: Information about using `ACYCLIC` to disallow the repetition of nodes in a path.
= Acyclic patterns
:table-caption!:
:page-role: new-neo4j-2026.03 cypher-25-only

> By default, Cypher only prevents relationships to be repeated in a `MATCH` result, not nodes (see xref:patterns/unique-relationship-matching.adoc[] for more information).

Reviewer: ```suggestion
By default, Cypher only prevents relationships being repeated in a `MATCH` result, not nodes (see xref:patterns/unique-relationship-matching.adoc[] for more information).
```

The `ACYCLIC` keyword adds a further restriction by not allowing nodes to be repeated within a path, though they can still be repeated across paths enabling the formation of equijoins.

The `ACYCLIC` keyword must be specified before a path pattern:

.Query using `ACYCLIC`
[source, cypher]
----
MATCH p = ACYCLIC (:Router {name: 'A'})-[:LINK]-+(:Router {name: 'Z'})
RETURN p
----

[NOTE]
`ACYCLIC` is a xref:appendix/gql-conformance/index.adoc[GQL conformant] path mode.
Path modes are levels of constraints that determine whether nodes or relationships can appear more than once in a path pattern match.
Cypher also supports explicitly specifying the GQL path modes `TRAIL` and `WALK` though these do not alter the constraints already imposed by `DIFFERENT RELATIONSHIPS` or `REPEATABLE ELEMENTS` respectively.

> For more information, see xref:patterns/reference.adoc#path-modes[Syntax and semantics -> Path modes]

Reviewer 2: ```suggestion
For more information, see xref:patterns/reference.adoc#path-modes[Syntax and semantics -> Path modes].
```
